GVFS/GVFS.Platform.Mac/ProjFSKext.cs

```diff
@@ -44,16 +50,21 @@ public bool IsSupported(string normalizedEnlistmentRootPath, out string warning,
 
         public string FlushLogs()
         {
-            throw new NotImplementedException();
+            Directory.CreateDirectory(this.LogsFolderPath);
+            ProcessResult logShowOutput = ProcessHelper.Run("log", args: "show --predicate \"subsystem contains \'org.vfsforgit\'\"", redirectOutput: true);
+            File.WriteAllText(Path.Combine(this.LogsFolderPath, "PrjFSKext.log"), logShowOutput.Output);
+
+            // Unlike FlushLogs on Windows, we will always produce a log file, even if it's empty if there is no results in the log subsystem.
+            return string.Empty;

```

Comment:
What does this function always return string.empty, I guess a void isn't an option here?

Reply:
The FlushLogs method that we expose in the Kernel Driver interface is rather... strange, likely because it was designed around how PrjFlt handles logs on Windows. The Windows flavor of this method (in `ProjFSFilter.cs`) potentially returns a string of errors containing exceptions on why we failed to extract the logs from PrjFlt for consumption in Diagnose. 

I could rework the interface as part of this change, but the complexity around how PrjFlt logs will have to be handled somewhere so I went for the cheapest fix.

I'm happy to rework if people have opinions on this!

/cc: @wilbaker

Reply:
@nickgra I think it's fine as it, if we can clarify a bit in the comment that would be great!

Reply:
> I'm happy to rework if people have opinions on this!

@nickgra how about updating this to our usual TryXXX pattern:

```
public bool TryFlushLogs(out string error)
```